File: uploaddemo/src/business/galary/java/com/jw/galary/img/bean/ImageItem.java
package com.jw.galary.img.bean;

import android.os.Parcel;
import android.os.Parcelable;

import com.jw.galary.base.BaseItem;

import java.io.Serializable;

public class ImageItem extends BaseItem implements Serializable, Parcelable {
    public int width;
    public int height;
    public long addTime;
    public static final Creator<ImageItem> CREATOR = new Creator<ImageItem>() {
        public ImageItem createFromParcel(Parcel source) {
            return new ImageItem(source);
        }

        public ImageItem[] newArray(int size) {
            return new ImageItem[size];
        }
    };

    protected ImageItem(Parcel in) {
        setName(in.readString());
        setPath(in.readString());
        setSize(in.readLong());
        width = in.readInt();
        height = in.readInt();
        setMimeType(in.readString());
        addTime = in.readLong();
    }

    public int describeContents() {
        return 0;
    }

    public boolean equals(Object o) {
        if (o instanceof ImageItem) {
            ImageItem item = (ImageItem) o;
            return getPath().equalsIgnoreCase(item.getPath());
        } else {
            return super.equals(o);
        }
    }

    public ImageItem() {
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeString(getPath());
        dest.writeLong(getSize());
        dest.writeInt(width);
        dest.writeInt(height);
        dest.writeString(getMimeType());
        dest.writeLong(addTime);
    }
}

File: library/src/main/java/com/jw/library/model/ImageItem.kt
package com.jw.library.model

import android.os.Parcel
import android.os.Parcelable

/**
 * 图片类
 * @property width Int
 * @property height Int
 * @property addTime Long
 * @property orientation Int    图片方向
 */
class ImageItem : BaseItem {
    var width: Int = 0
    var height: Int = 0
    var addTime: Long = 0
    var orientation: Int = 0

    constructor()

    constructor(parcel: Parcel) : super(parcel) {
        width = parcel.readInt()
        height = parcel.readInt()
        addTime = parcel.readLong()
        orientation = parcel.readInt()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        super.writeToParcel(parcel, flags)
        parcel.writeInt(width)
        parcel.writeInt(height)
        parcel.writeLong(addTime)
        parcel.writeInt(orientation)
    }

    companion object CREATOR : Parcelable.Creator<ImageItem> {
        override fun createFromParcel(parcel: Parcel): ImageItem {
            return ImageItem(parcel)
        }

        override fun newArray(size: Int): Array<ImageItem?> {
            return arrayOfNulls(size)
        }
    }
}
